python/cudf/cudf/io/parquet.py

```diff
@@ -534,6 +536,146 @@ def write_to_dataset(
     return metadata
 
 
+_BIN_OPS = {
+    "==": plc_expr.ASTOperator.EQUAL,
+    "!=": plc_expr.ASTOperator.NOT_EQUAL,
+    "<": plc_expr.ASTOperator.LESS,
+    "<=": plc_expr.ASTOperator.LESS_EQUAL,
+    ">": plc_expr.ASTOperator.GREATER,
+    ">=": plc_expr.ASTOperator.GREATER_EQUAL,
+}
+
+
+def translate_filters_to_ast(
+    filters: list[list[tuple[str, str, Any]]],
+) -> plc_expr.Expression:
+    """
+    Convert a filter expression in Disjunctive Normal Form (DNF) to a pylibcudf AST.
+
+    Parameters
+    ----------
+    filters : list[list[tuple[str, str, Any]]]
+        A filter expression in DNF.
+        DNF is represented as a list of OR-ed clauses, where each clause is a list of
+        AND-ed predicates. Each predicate is a tuple of the form:
+            (column_name, operator, value)
+
+    Returns
+    -------
+    plc.expressions.Expression
+        A pylibcudf AST expression representing the filter.
+
+    Notes
+    -----
+    This function supports the following operators:
+      - Binary comparisons: ==, !=, <, <=, >, >=
+      - Membership tests: "in", "not in"
+      - Null checks: "is None", "is not None"
+
+    The translation is performed by recursively reducing:
+      - Each clause's predicates into a conjunction (AND)
+      - All clauses into a disjunction (OR) of those conjunctions
+    """
+
+    def _raise_for_unsupported_scalar_types(val: Any):
+        if not (
+            isinstance(val, (int, float))
+            or isinstance(val, (datetime.date, datetime.timedelta))
+            or isinstance(val, str)
+        ):
+            return val
+        raise NotImplementedError(
+            "Only numeric, string, or timestamp/duration scalars are accepted"
+        )
+
+    def make_expr(col: str, op: str, val: Any) -> plc_expr.Expression:
+        col_ref = plc_expr.ColumnNameReference(col)
+
+        if op in _BIN_OPS:
+            return plc_expr.Operation(
+                _BIN_OPS[op],
+                col_ref,
+                plc_expr.Literal(
+                    plc.Scalar.from_py(
+                        _raise_for_unsupported_scalar_types(val)
+                    )
+                ),
```

Comment:
Maybe abstract this `Literal` construction into a separate function.

Reply:
Done